pkg/api/payer/nodeSelector.go

```diff
@@ -39,22 +40,46 @@ func (s *StableHashingNodeSelectorAlgorithm) GetNode(
 		return 0, err
 	}
 
-	if len(nodes) == 0 {
-		return 0, errors.New("no available nodes")
+	// Flatten banlist
+	banned := make(map[uint32]struct{})
+	for _, list := range banlist {
+		for _, id := range list {
+			banned[id] = struct{}{}
+		}
 	}
 
-	// Sort nodes to ensure stability
-	sort.Slice(nodes, func(i, j int) bool { return nodes[i].NodeID < nodes[j].NodeID })
+	// Filter out banned, disabled, or non-API nodes
+	var availableNodes []registry.Node
+	for _, node := range nodes {
+		if node.IsDisabled || !node.IsApiEnabled {
+			continue
+		}
```

Comment:
I don't think that this is the right place to do this. reg.GetNodes() already does a bunch of pre-filtering. Why dont we do it there so its all in one place? Look at `func (s *SmartContractRegistry) refreshData() error`